src/net/NodeHttpClient.ts

```diff
@@ -1,12 +1,25 @@
-import * as http from 'http';
-import * as https from 'https';
+// It's important to do a default import here, rather thane
+// `import * as http from 'http'` because the latter will
+// use a "Module Namespace Exotic Object" which is immune to
+// monkey-patching, whereas http.default (in an ES Module context)
+// will resolve to the same thing as require('http'), which is
+// monkey-patchable. We care about this because users in their test
+// suites might be using a library like "nock" which relies on the ability
+// to monkey-patch and intercept calls to http.request.
+
+// Get rid of this type error Module '"http"' has no default export.ts(1192)
```

Comment:
This comment is defunct. I think a comment lower down will be more appropriate.
```suggestion
// It's important to do a default import here, rather thane
// `import * as http from 'http'` because the latter will
// use a "Module Namespace Exotic Object" which is immune to
// monkey-patching, whereas http.default (in an ES Module context)
// will resolve to the same thing as require('http'), which is
// monkey-patchable. We care about this because users in their test
// suites might be using a library like "nock" which relies on the ability
// to monkey-patch and intercept calls to http.request.

// Get rid of this type error Module '"http"' has no default export.ts(1192)
```